### ISSUE TITLE
Kafka 0.10 support

### ISSUE THREAD

**Issue by panda87** [braedon/prometheus-kafka-consumer-group-exporter#7]
Hi

Im trying to use this great exporter and Im not getting anything but `kafka_topic_highwater`
I have groups in __consumer_group topic.

Maybe it's versions issue?
Im using kafka 0.10.2.1

Thanks
D.

**Comment by braedon** [comment]
Hi @panda87, I haven't tried that version of Kafka specifically, but use 0.10.1.0 - I wouldn't expect there to be any problems with other versions in the 0.10 series.

Are your consumers actively committing? By default the exporter only reads new messages from the `__consumer_group` topic (reading the topic history would be wasteful and produce misleading metrics on startup in most cases).

You can use the `--from-start` flag (`-s`) to tell the exporter to read from the start of the `__consumer_group` topic if you need to.

Let me know how you get on!

**Comment by panda87** [comment]
Hi

I opened another issue in other [repo](https://github.com/kawamuray/prometheus-kafka-consumer-group-exporter/issues/24) and we agreed that Kafka team changed the output in version 0.10.2.1 (maybe before).

I agree with you that I need to check that with running consumer to see the latest change, but anyway I think that if the output has changed, the offset exporter will result issue to present the metrics.

**Comment by braedon** [comment]
Hi @panda87, I just tested the exporter on Kafka 0.10.2.1 and it seems to work fine for me - all expected metrics are there and being updated.

Let me know if you're still having trouble after testing with active consumers (or with `-s` on). Thanks!

**Comment by panda87** [comment]
Thanks @braedon for your time and effort! I will test that soon

**Comment by stijndehaes** [comment]
I tested this library with kafka 0.10.2.1 with the docker-compose file at the end of the mesage. I used kafka cat with the following commands to post some messages:
`kafkacat -b localhost:9092 -P -t test`
And consumed with the following command:
`kafkacat -b localhost:9092 -C -G group1 test`
I saw the kafka_topic_highwater metrics change but no other metrics where reported. THe library was started with the following command:
`prometheus-kafka-consumer-group-exporter -b localhost:9092`
after installing it with pip3.

Any idea why the other metrics aren't reported?

```
version: '2'
services:
  kafka:
    image: wurstmeister/kafka:0.10.2.1
    ports:
    - 9092:9092
    - 1999:1999
    environment:
    - KAFKA_ADVERTISED_HOST_NAME=0.0.0.0
    - KAFKA_ZOOKEEPER_CONNECT=zookeeper:2181
    - KAFKA_ADVERTISED_PORT=9092
    depends_on:
    - zookeeper
  zookeeper:
    image: zookeeper:3.4.10
    ports:
    - 2181:2181
    environment:
    - ZOO_MY_ID=1
    - ZOO_SERVERS=server.1=zookeeper:2888:3888
```

**Comment by braedon** [comment]
Hi @stijndehaes, that's the same kafka docker image I used for my test (I'm using the latest wurstmeister zookeeper image, but that shouldn't affect this).

I tried it again, however, and used kafkacat to consume messages. Once again, `kafka_consumer_group_commits` and `kafka_consumer_group_offset` metrics show up fine for me. It did seem to take a while for kafkacat to actually do its first commit - not sure what its commit strategy is? You might need to commit more messages, or wait longer for the consumer to commit.

In case it makes a difference, this is the kafkacat version Ubuntu gave me:
```
$ apt show kafkacat
Package: kafkacat
Version: 1.3.0-1
...
$ apt show librdkafka1
Package: librdkafka1
Version: 0.9.3-1
...
```

I'd recommend pointing the exporter at a production cluster and using the `-s` flag to see how it works with your actual consumers.

If you still have issues, it might pay to confirm you have the latest version of the exporter, and update if not:
```
$ pip3 show prometheus-kafka-consumer-group-exporter
Name: prometheus-kafka-consumer-group-exporter
Version: 0.2.0
...
```
```
pip3 install -U prometheus-kafka-consumer-group-exporter
```
Testing the docker version of the exporter could also help eliminate version issues.
```
sudo docker pull braedon/prometheus-kafka-consumer-group-exporter:0.2.0
```
